cosima_cookbook/database.py

```diff
@@ -332,6 +332,14 @@ def index_experiment(experiment_dir, session=None, client=None, update=False):
     # make all files relative to the experiment path
     files = [str(Path(f).relative_to(expt_path)) for f in files]
 
+    missing_files = {}
+    if prune:
+        # create a dictionary of all entries that are missing
+        for f in expt.ncfiles:
+            missing_files[f.ncfile] = f
+        for f in files:
+            del(missing_files[f])
```

Comment:
`KeyError` from here, guess you'd need to check `f` in `missing_files` first

Reply:
Tah!

Reply:
Wrapping in a `try/except` blocks fixes the error, but it seems wasteful to figure out which files are new and exclude them, to reproduce the same in reverse in the subsequent code blocks when `update=True`.

When `update=False` does it re-read all the metadata for all files, even ones already in the DB?

Reply:
To answer myself, no it doesn't, because it can't get that far if `update=True` due to this check:
https://github.com/COSIMA/cosima-cookbook/blob/master/cosima_cookbook/database.py#L319-L326
which makes the check for `update` here superfluous I think
https://github.com/COSIMA/cosima-cookbook/blob/master/cosima_cookbook/database.py#L335